### ISSUE TITLE
test for bash scripts 

### ISSUE THREAD

**Issue by heikef** [qmcurrents/gimic#208]
This issue is to discuss if there is a way to add one or two tests for the bash scripts. As it is only Maria is developing the scripts but for future usage it would be good to have some tests to assure quality. 
We have to decide if we want to go for this or if we consider this to be too much work. 
Please share comments/opinions on this topic @mariavd @bast

**Comment by bast** [comment]
If we don't test them, then what will happen at some point is that stuff will stop working and we will not notice. We will change keywords and defaults and it will be very easy to break the scripts without noticing if they are not somehow automatically tested.

**Comment by heikef** [comment]
I agree. Perhaps it depends if we want to keep them for the future or not. Maria put a lot of work in setting them up. All this work should not be lost. Is it possible to set up tests for bash scripts like the ones we have? They are rather complex.

**Comment by bast** [comment]
For sure possible to test. And if they are too complex, we should simplify.

**Comment by mariavd** [comment]
I have thought about this before but I never brought it up. I think the simplest way to test the current profile script is to call it with selected parameters. The first check will be to test if it produces the input files as expected. Then the script be left to execute the calculations for the first and last slice and the results checked for consistency. 

I am willing to do it but most likely not before the end of November.

**Comment by mariavd** [comment]
As to complexity - well, that was intentional. As you know, I'm lazy to setup planes manually and edit input files by hand. I'd rather have the script do it for me. I wanted them to be interactive, and that's more than complicated than just echo, read, and a couple of sed lines. 

I admit that the way they evolved, it would have been better to have all this in python. But in the beginning, a couple of years ago, I just wanted to improve on several short bash scripts that I was given. Then ideas kept on coming when I was annoyed at one thing or another when setting up my calculations. Now I am sourcing a file with bash functions to be used in the main file, which by itself is like 500 lines...

**Comment by heikef** [comment]
Good. Let's agree that we want to add tests for the scripts and keep this issue open.

**Comment by bast** [comment]
So the tests are probably not urgent and the motivation to add them would be to secure that they don't get broken. We might consider having a side-repo with the helper scripts but also then they should be somehow tested I believe.

**Comment by mariavd** [comment]
You @bast and @heikef as the main developers can decide how/if you prefer to have these scripts as part of the official gimic repo. I will prepare the tests sometimes.

**Comment by bast** [comment]
If they are useful for you they will likely be useful for others. For me it wasn't a problem yet to have them in the main repo. When I suggest to change some defaults or naming, I always do a `git grep` and hope I catch everything.

The advantage of putting things like this in a side repo (e.g. `contributed-scripts`) is that the main repo is "cleaner" and less things to keep track of. But the clear disadvantage would be "out of sight out of mind" and breakage would be guaranteed.

So long story short, I have no strong opinion on this.

**Comment by patrashantigopal** [comment]
I am having a problem in running $ make test:

dabu@RES-SHANTIPA:/mnt/c/Users/Shantigp/projects/gimic-master/build$ make test
Running tests...
Test project /mnt/c/Users/Shantigp/projects/gimic-master/build
      Start  1: benzene/integration-gauss
 1/24 Test  #1: benzene/integration-gauss .........***Failed   28.46 sec
      Start  2: benzene/integration-lobatto
 2/24 Test  #2: benzene/integration-lobatto .......***Failed    0.99 sec
      Start  3: benzene/vectors
 3/24 Test  #3: benzene/vectors ...................***Failed    1.20 sec
      Start  4: benzene/2d
 4/24 Test  #4: benzene/2d ........................***Failed    1.14 sec
      Start  5: benzene/3d
 5/24 Test  #5: benzene/3d ........................***Failed    1.95 sec
      Start  6: benzene/keyword-magnet
 6/24 Test  #6: benzene/keyword-magnet ............***Failed    1.06 sec
      Start  7: benzene/3d-keyword-magnet
 7/24 Test  #7: benzene/3d-keyword-magnet .........***Failed    2.09 sec
      Start  8: benzene/2d-keyword-magnet
 8/24 Test  #8: benzene/2d-keyword-magnet .........***Failed    1.21 sec
      Start  9: benzene/keyword-spacing
 9/24 Test  #9: benzene/keyword-spacing ...........***Failed    1.08 sec
      Start 10: benzene/keyword-rotation
10/24 Test #10: benzene/keyword-rotation ..........***Failed    1.01 sec
      Start 11: benzene/keyword-rotation_origin
11/24 Test #11: benzene/keyword-rotation_origin ...***Failed    1.04 sec
      Start 12: benzene/keyword-radius
12/24 Test #12: benzene/keyword-radius ............***Failed    1.22 sec
      Start 13: benzene/int-grid-bond-even
13/24 Test #13: benzene/int-grid-bond-even ........***Failed    1.02 sec
      Start 14: benzene/int-cdens
14/24 Test #14: benzene/int-cdens .................***Failed    1.23 sec
      Start 15: benzene/diamag-off
15/24 Test #15: benzene/diamag-off ................***Failed    1.19 sec
      Start 16: benzene/paramag-off
16/24 Test #16: benzene/paramag-off ...............***Failed    1.30 sec
      Start 17: benzene/giao-test
17/24 Test #17: benzene/giao-test .................***Failed    1.31 sec
      Start 18: c4h4/integration
18/24 Test #18: c4h4/integration ..................***Failed    1.31 sec
      Start 19: c4h4/read-grid
19/24 Test #19: c4h4/read-grid ....................***Failed    1.28 sec
      Start 20: open-shell/3d
20/24 Test #20: open-shell/3d .....................***Failed    5.27 sec
      Start 21: open-shell/integration
21/24 Test #21: open-shell/integration ............***Failed    1.35 sec
      Start 22: benzene/valgrind-integral
22/24 Test #22: benzene/valgrind-integral .........***Failed   19.49 sec
      Start 23: benzene/valgrind-2d
23/24 Test #23: benzene/valgrind-2d ...............***Failed    3.49 sec
      Start 24: benzene/skip-jmod-integration
24/24 Test #24: benzene/skip-jmod-integration .....***Failed    1.38 sec

0% tests passed, 24 tests failed out of 24

Total Test time (real) =  89.35 sec

The following tests FAILED:
          1 - benzene/integration-gauss (Failed)
          2 - benzene/integration-lobatto (Failed)
          3 - benzene/vectors (Failed)
          4 - benzene/2d (Failed)
          5 - benzene/3d (Failed)
          6 - benzene/keyword-magnet (Failed)
          7 - benzene/3d-keyword-magnet (Failed)
          8 - benzene/2d-keyword-magnet (Failed)
          9 - benzene/keyword-spacing (Failed)
         10 - benzene/keyword-rotation (Failed)
         11 - benzene/keyword-rotation_origin (Failed)
         12 - benzene/keyword-radius (Failed)
         13 - benzene/int-grid-bond-even (Failed)
         14 - benzene/int-cdens (Failed)
         15 - benzene/diamag-off (Failed)
         16 - benzene/paramag-off (Failed)
         17 - benzene/giao-test (Failed)
         18 - c4h4/integration (Failed)
         19 - c4h4/read-grid (Failed)
         20 - open-shell/3d (Failed)
         21 - open-shell/integration (Failed)
         22 - benzene/valgrind-integral (Failed)
         23 - benzene/valgrind-2d (Failed)
         24 - benzene/skip-jmod-integration (Failed)
Errors while running CTest
Makefile:94: recipe for target 'test' failed
make: *** [test] Error 8

I shall be very grateful if someone can help me.

Best regards,
Shanti

**Comment by bast** [comment]
dear Shanti, I will help you debug these but can you please open a new issue? This issue is unrelated - but I also understand that it looks related.

**Comment by heikef** [comment]
The bash scripts are not under active development any more as I understood. Closing this issue. We can reopen it if needed be.
